pilot/pkg/security/authz/matcher/string.go

```diff
@@ -43,9 +48,17 @@ func StringMatcherRegex(regex string) *matcherpb.StringMatcher {
 // created string matcher, note the prefix is ignored if v is wildcard ("*").
 // The wildcard "*" will be generated as ".+" instead of ".*".
 func StringMatcherWithPrefix(v, prefix string) *matcherpb.StringMatcher {
+	// If environment variable AUTHZ_ENABLE_REGEX is set to ture, prefix "regex:"
```

Comment:
Also update the function comment to match the new behavior.

Reply:
also typo in `triger`

Reply:
Do you think the current comment explains well? Thanks.

Reply:
I mean update the function comment above in line 44 (just mention `regex:` will be interpreted as regular expression when the flag is enabled), not here. Actually you can remove the comments here.